inluxclient/query.go

```diff
@@ -0,0 +1,315 @@
+package influxclient
+
+import (
+	"encoding/base64"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// TableColumn holds flux query table column properties
+type TableColumn struct {
+	Name         string
+	DataType     string
+	IsGroup      bool
+	DefaultValue string
+}
+
+// String returns TableColumn string dump
+func (f *TableColumn) String() string {
+	return fmt.Sprintf("{name: %s, datatype: %s, defaultValue: %s, group: %v}", f.Name, f.DataType, f.DefaultValue, f.IsGroup)
+}
+
+const (
+	stringDatatype       = "string"
+	doubleDatatype       = "double"
+	boolDatatype         = "boolean"
+	longDatatype         = "long"
+	uLongDatatype        = "unsignedLong"
+	durationDatatype     = "duration"
+	base64BinaryDataType = "base64Binary"
+	timeDatatypeRFC      = "dateTime:RFC3339"
+	timeDatatypeRFCNano  = "dateTime:RFC3339Nano"
+)
+
+type parsingState int
+type parsingMode int
+
+const (
+	parsingStateNoTable parsingState = iota
+	parsingStateDataRow
+	parsingStateAnnotation
+	parsingStateNameRow
+	parsingStateError
+)
+
+const (
+	parsingModeTable parsingMode = iota
+	parsingModeRow
+)
+
+// QueryResults parses Flux query response stream.
+// Walking though the result set is done by repeatedly calling NextTable() and NextRow() until return false.
+// NextRow() can be also called initially, to advance straight to the first row of the first table.
+// Actual flux table schema (columns with names, data types, etc) is returned by Columns() method.
+// Data are acquired by Values() or by ValueByName() functions.
+// Preliminary end can be caused by an error, so when NextTable() or NextRow() return false, check Err() for an error.
+// Reader is automatically closed at the end reading or in case of an error.
+type QueryResults struct {
+	io.Closer
+	csvReader *csv.Reader
+	columns   []TableColumn
+	values    []interface{}
+	err       error
+	lastRow   []string
+	names     map[string]int
```

Comment:
```suggestion
        // columnIndexes maps column names to their indexes.
	columnIndexes     map[string]int
```
I think it's nicer to name the map after the values rather than the keys.